packages/integration/src/github/SingleInstanceGithubCredentialsProvider.ts

```diff
@@ -267,15 +260,24 @@ export class SingleInstanceGithubCredentialsProvider
    */
   async getCredentials(opts: { url: string }): Promise<GithubCredentials> {
     const parsed = parseGitUrl(opts.url);
+    const gitHubConfig = this.integrations.github.byUrl(opts.url)?.config;
+    if (!gitHubConfig) {
+      throw new Error(
+        `There is no GitHub integration that matches ${opts.url}. Please add a configuration for an integration.`,
+      );
+    }
+
+    const githubAppCredentialsMux = new GithubAppCredentialsMux(gitHubConfig);
```

Comment:
We won't want to instantiate this every time, it contains a cache that stores information about installations so that we don't need to hit the GH API too much. You'd rather want to take the list of integration configs and create a mapping upfront

Reply:
Done